network/p2p/gossip/metrics.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	ioLabel    = "io"
+	sentIO     = "sent"
+	receivedIO = "received"
+
+	typeLabel  = "type"
+	pushType   = "push"
+	pullType   = "pull"
+	unsentType = "unsent"
+	sentType   = "sent"
+
+	// dropped indicate that the gossipable element was not added to the set due to some reason.
+	// for sent message, we'll use notReceive below.
+	droppedLabel = "dropped"
+
+	droppedMalformed = "malformed"
+	DroppedDuplicate = "duplicate"
+	droppedNot       = "not"
+	notReceive       = "not_receive"
+)
+
+var (
+	ioTypeDroppedLabels = []string{ioLabel, typeLabel, droppedLabel}
+	sentPushLabels      = prometheus.Labels{
+		ioLabel:      sentIO,
+		typeLabel:    pushType,
+		droppedLabel: notReceive,
+	}
+	sentPullLabels = prometheus.Labels{
+		ioLabel:      sentIO,
+		typeLabel:    pullType,
+		droppedLabel: notReceive,
+	}
+	typeLabels   = []string{typeLabel}
+	unsentLabels = prometheus.Labels{
+		typeLabel: unsentType,
+	}
+	sentLabels = prometheus.Labels{
+		typeLabel: sentType,
+	}
+)
+
+// Metrics that are tracked across a gossip protocol. A given protocol should
+// only use a single instance of Metrics.
+type Metrics struct {
+	count                   *prometheus.CounterVec
+	bytes                   *prometheus.CounterVec
+	tracking                *prometheus.GaugeVec
+	trackingLifetimeAverage prometheus.Gauge
+	topValidators           *prometheus.GaugeVec
+
+	labelsLock        sync.Mutex
+	gossipablesLabels map[ids.ID]string
```

Comment:
Do we even need this map + lock? Can't we just update the drop metrics similarly to how we're updating the existing metrics?